src/sql/workbench/parts/dataExplorer/electron-browser/nodeActions.contribution.ts

```diff
@@ -18,24 +19,31 @@ MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
 	when: NodeContextKey.IsConnected
 });
 
+// For Database nodes under a Server in the SQL Servers folder
 MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
 	group: 'connection',
 	order: 2,
 	command: {
 		id: NEW_QUERY_COMMAND_ID,
 		title: localize('newQuery', 'New Query')
 	},
-	when: NodeContextKey.IsConnectable
+	when: ContextKeyExpr.and(
+		NodeContextKey.IsConnectable,
+		new ContextKeyRegexExpr('viewItem', /Database|azure.resource.itemType.database|azure.resource.itemType.databaseServer/))
```

Comment:
Can you just make it a regex string that searches the whole string for "server" or "database"? Something like `.+(server|database)` that way we don't need to have azure strings in our core code.

Reply:
Sure - I'll do that for now. TBH though this whole thing should probably be looked at if we're really concerned about extending it to things beyond the Azure tree (such as CMS). Having some sort of nodeType on here like OE does would be a good start.

Reply:
Well it isn't pretty but it at least doesn't have the azure strings in it. The main problem is that we have no consistent standard for naming these things - the root nodes have a very different naming scheme than the subnodes and some of those subnodes have database or server in their name (such as ServerLevelLogin) that we don't want to have these menu items on. 

---
In reply to: [274688271](https://github.com/Microsoft/azuredatastudio/pull/4996#discussion_r274688271) [](ancestors = 274688271)